starter-code/public/scripts/article.js

```diff
@@ -3,128 +3,156 @@ var app = app || {};
-  .then(callback);
-};
-
-Article.prototype.insertRecord = function(callback) {
-  // REVIEW: Why can't we use an arrow function here for .insertRecord()??
-  $.post('/articles', {author: this.author, authorUrl: this.authorUrl, body: this.body, category: this.category, publishedOn: this.publishedOn, title: this.title})
-  .then(console.log)
-  .then(callback);
-};
-
-Article.prototype.deleteRecord = function(callback) {
-  $.ajax({
-    url: `/articles/${this.article_id}`,
-    method: 'DELETE'
-  })
-  .then(console.log)
-  .then(callback);
-};
-
-Article.prototype.updateRecord = function(callback) {
-  $.ajax({
-    url: `/articles/${this.article_id}`,
-    method: 'PUT',
-    data: {
-      author: this.author,
-      authorUrl: this.authorUrl,
-      body: this.body,
-      category: this.category,
-      publishedOn: this.publishedOn,
-      title: this.title,
-      author_id: this.author_id
-    }
-  })
-  .then(console.log)
-  .then(callback);
-};
+
+(function(module) {
+  function Article(rawDataObj) {
+    /* REVIEW: In lab 8, we explored a lot of new functionality going on here. Let's re-examine
+    the concept of context.
+    Normally, "this" inside of a constructor function refers to the newly instantiated object.
+    However, in the function we're passing to forEach, "this" would normally refer to "undefined"
+    in strict mode. As a result, we had to pass a second argument to forEach to make sure our "this"
+    was still referring to our instantiated object.
+    One of the primary purposes of lexical arrow functions, besides cleaning up syntax to use fewer
+    lines of code, is to also preserve context. That means that when you declare a function using
+    lexical arrows, "this" inside the function will still be the same "this" as it was outside
+    the function.
+    As a result, we no longer have to pass in the optional "this" argument to forEach!*/
+    Object.keys(rawDataObj).forEach(key => this[key] = rawDataObj[key]);
+  }
+
+  Article.all = [];
+
+  Article.prototype.toHtml = function() {
+    var template = Handlebars.compile($('#article-template').text());
+
+    this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+    this.publishStatus = this.publishedOn ? `published ${this.daysAgo} days ago` : '(draft)';
+    this.body = marked(this.body);
+
+    return template(this);
+  };
+
+  Article.loadAll = rows => {
+    rows.sort((a,b) => (new Date(b.publishedOn)) - (new Date(a.publishedOn)));
+
+    // DONE: Refactor this forEach code, by using a `.map` call instead, since what we are trying to accomplish
+    // is the transformation of one collection into another. Remember that we can set variables equal to the result
+    // of functions. So if we set a variable equal to the result of a .map, it will be our transformed array.
+    // There is no need to push to anything.
+
+    rows.map(ele => Article.all.push(new Article(ele)));
```

Comment:
This is improper use of `.map()`. You do not need to use `.push()` to fill `Article.all` since `.map()` returns a new array.